plugin/builtin/buildbaron/build_baron.go

```diff
@@ -130,6 +150,44 @@ type searchReturnInfo struct {
 	Search string                  `json:"search"`
 }
 
+// raceSuggesters returns the JIRA ticket results from the altEndpoint suggester if it returns
+// within its configured interval, and returns the JIRA ticket results from the fallback suggester
+// otherwise.
+func raceSuggesters(fallback, altEndpoint suggester, t *task.Task) ([]thirdparty.JiraTicket, error) {
+	type result struct {
+		Tickets []thirdparty.JiraTicket
+		Error   error
+	}
+
+	// thirdparty/jira.go and thirdparty/http.go do not expose an API that accepts a context.Context.
+	fallbackCtx := context.TODO()
+	fallbackChan := make(chan result)
+	go func() {
```

Comment:
at the very least bare adhoc goroutines should have a recovery handler, but it'd be cool if we could avoid it entirely. 

what's the timeout on this? will the loadbalancer time us out before then anyway?

Reply:
There isn't currently a timeout on the API server calling out to JIRA and there still won't be one after my changes. What would a recovery handler look like for this goroutine, or how could it be avoided entirely?

Reply:
@tychoish Our rest routes all have panic handlers, is one here actually necessary?

Reply:
I think there is, in effect, because the load balancer will timeout incoming requests, leaving this thread spinning (for a potentially long time) even after there's no one waiting for the request. We should set reasonable timeouts for this to avoid giving jira a mechanism to bring down evergreen web servers.

Reply:
Per our in-person discussion, this goroutine is only going to run for as long as talking to JIRA would otherwise have taken. That is to say, I don't think we've changed the impact JIRA being slow can have on this particular endpoint. The ultimate goal once the TIG team's deployment model for the Build Baron suggestions stabilizes would be to serve JIRA results as a fallback on our end and remove the need for the `raceSuggesters()` function.